lib/tasks_private/terraform/provision.rake

```diff
@@ -0,0 +1,91 @@
+require "json"
+require "byebug"
+require 'yaml'
+require 'logger'
+require "byebug"
+
+begin
+  require 'ruby-terraform'
+rescue LoadError
+  raise("unable to provision vm on Power VC as ruby terraform gem is not available")
+end
+
+namespace :provision do
+  task :check => :environment do
+    desc "check terraform is installed"
+    system("command -v terraform > /dev/null") or raise("install terraform to provision powervc vm")
+  end
+
+  task :apply => :check do
+    desc "provision a vm"
+    cdir = ENV['cwd'] || Dir.pwd
+    tdir = __dir__
+    config = setup(cdir, tdir)
+    plan(config)
+    apply(cdir, tdir)
+  end
+
+  task :destroy => :check do
+    desc "delete a vm"
+    cdir = ENV['cwd'] || Dir.pwd
+    tdir = __dir__
+    config = setup(cdir, tdir)
+    destroy(cdir, config)
+  end
+end
+
+def setup(cdir, tdir)
```

Comment:
So as not to infect the global namespace, as this rake task will be loaded all the time, prefer to put these method inside of a namespace, or put it into a separate rake task helper ruby file that is loaded on demand.